### PR TITLE
Update summary quantity on combination list refresh

### PR DESCRIPTION
**Pull request by zuk3975**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.1.x
| Description?      | https://github.com/PrestaShop/PrestaShop/issues/29349
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | https://github.com/PrestaShop/PrestaShop/issues/29349
| Fixed ticket?     | https://github.com/PrestaShop/PrestaShop/issues/29349
| Related PRs       | 
| Sponsor company   |

### CODE REVIEW

**Review by matthieu-rolland — DISMISSED** [review]

File: src/Adapter/Product/Stock/Repository/StockAvailableMultiShopRepository.php [hunk 1]
```diff
@@ -200,12 +200,10 @@ public function delete(ProductId $productId, ShopId $shopId): void
      *
      * @return StockId
      *
-     * @throws CoreException
      * @throws StockAvailableNotFoundException
      */
     public function getStockIdByCombination(CombinationId $combinationId, ShopId $shopId): StockId
     {
-        //@todo: add shop conditions based on shop group sharing stock or not. like in ProductCombinationQueryBuilder
```
Comment (zuk3975) [inline src/Adapter/Product/Stock/Repository/StockAvailableMultiShopRepository.php]: unrelated.
Just saw its already fixed in line 216 so I removed this todo

**Review by kpodemski — DISMISSED** [review]

**Comment by zuk3975** [comment]
need approvals after rebase

**Review by lartist — DISMISSED** [review]

**Review by lartist — DISMISSED** [review]

File: admin-dev/themes/new-theme/js/pages/product/edit/summary-quantity-updater.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+import {getProductQuantity, QuantityResult} from '@pages/product/service/product';
+import ProductMap from '@pages/product/product-map';
+import {EventEmitter} from '@components/event-emitter';
+import ProductEventMap from '@pages/product/product-event-map';
+
+export default class SummaryQuantityUpdater {
```
Comment (jolelievre) [inline admin-dev/themes/new-theme/js/pages/product/edit/summary-quantity-updater.ts:31]:
```suggestion
export default class HeaderSummaryUpdater {
```

Comment (jolelievre) [reply admin-dev/themes/new-theme/js/pages/product/edit/summary-quantity-updater.ts:31]: For now, only the quantity field is updated but I'm almost sure at some point the other fields will be asked to be updated as well (prices, references, and so on). So we'd better keep a more generic naming for this one that will be used in the future

File: admin-dev/themes/new-theme/js/pages/product/service/product.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import Router from '@components/router';
+
+const router = new Router();
+const {$} = window;
+
+export interface QuantityResult {
+  quantity: number,
+}
+
+export const getProductQuantity = async (productId: number, shopId: number): Promise<QuantityResult> => $.get(
```
Comment (jolelievre) [inline admin-dev/themes/new-theme/js/pages/product/service/product.ts]: If you can use the native `fetch` method instead of JQuery it would be nice

**Review by jolelievre — COMMENTED** [review]
Thanks @zuk3975 

only some minor comments for improvement.

File: admin-dev/themes/new-theme/js/pages/product/edit/summary-quantity-updater.ts (hunk 2 above)
Comment (zuk3975) [reply admin-dev/themes/new-theme/js/pages/product/edit/summary-quantity-updater.ts:31]: we already have a PriceSummary that also updates price in header, I had an idea to merge them into some single SummaryUpdater but the price one is quite complicated, so I didn't go for it :thinking: Can you check that one and reply what you think about it?

**Review by jolelievre — DISMISSED** [review]
Thanks @zuk3975 

suggestions not blocking, if you get feedback to fix by QA though it might be worth integrating them.

File: admin-dev/themes/new-theme/js/pages/product/edit/summary-quantity-updater.ts (hunk 2 above)
Comment (zuk3975) [reply admin-dev/themes/new-theme/js/pages/product/edit/summary-quantity-updater.ts:31]: as discussed in private, it seems that one doesn't actually update the header (but the prices in price tab summary), but as you said now its not worth renaming this service :smile:

**Review by jolelievre — APPROVED** [review]
Thanks @zuk3975

**Review by matthieu-rolland — APPROVED** [review]

**Comment by AureRita** [comment]
Hi @zuk3975 

Thank you for this PR, I tested it and I seems to works, as we can see here : 

[Untitled_ Mar 3, 2023 5_07 PM.webm](https://user-images.githubusercontent.com/100699445/222769610-d6bb6816-778d-4417-b15f-224bf2ff8bb6.webm)

We can see that it's correctly present on the second shop, but we didn't have the total's quantity of all shop. I know this PR doesn't fix it but it can be a sequel of this PR

**Comment by zuk3975** [comment]
thanks :+1:

**Pull request merged**